src/dmd/dsymbol.d

```diff
@@ -1641,6 +1641,42 @@ public:
         }
     }
 
+
+    /*****************************************
+     * Returns: the symbols whose members have been imported, i.e. imported modules
+     * and template mixins.
+     *
+     * See_Also: importScope
+     */
+    final Dsymbols* getImportedScopes() nothrow
+    {
+        return importedScopes;
+    }
+
+    /*****************************************
+     * Returns: the visibilities array associated to each imported scopes. The
+     * length of the array match with the imported modules array.
+     *
+     * See_Also: getImportedScopes, getImportVisibilitiesSlice
+     */
+    final Visibility.Kind* getImportVisibilities() nothrow
```

Comment:
I don't see a reason to expose this function. I suggest keeping only `getImportVisibilitiesSlice` and renaming it to `getImportVisibilities`

Reply:
I added it for people who might use `libdmd` as `extern(C++)`.

Reply:
Yes, that's clear. But we shouldn't add new APIs without definite use cases. (The YAGNI principle.) Given that you're working on a linter based on libdmd, it's clear that deficiencies in the `extern(D)` interface should be fixed, if they're blocking your work. But on the other hand, I think it's highly unlikely that there would be new users of the `extern(C++)` interface, apart from GDC and LDC.
Please note, that there has been a lot of effort put into reducing the C++ API surface to the bare minimum (and a lot more can be done still).

But in the absence of such users, we shouldn't increase the C++ API surface.

Reply:
> But in the absence of such users, we shouldn't increase the C++ API surface.

Makes sense to me, I don't see a usecase either.